### PR TITLE
Adding an HTML reporter.

### PR DESCRIPTION
**Pull request by yagni**
Since the post limit got upped to 50mb, that freed me up to write an HTML report based on blanket's own qunit default reporter. This includes branch coverage and displays both line and branch coverage on the original source. Module support is also included. As such, I had to reinstate sending source information in blanket-reporter.js.

I also updated blanket-reporter.js so that the innermost array is now an object: because blanket's branch properties "consequence" and "alternate" were stuck on an array and thus not serialized across the wire, those had to be put in an object. Updated JSON reporter to take this into account.

Lastly, I updated test fixture data using blanket output from https://github.com/ember-cli/ember-cli-todos since I couldn't find the code that the existing fixture data was generated from, and it didn't have blanket consequence and alternate properties. The HTML reports match what I'm seeing out of blanket, and the other reports look reasonable based on their previous versions and the blanket report from ember-cli-todos.

### CODE REVIEW

**Comment by sglanzer** [comment]
This is great - we'll probably need a few days to get to a review, but this is a very nice addition, thank you!

File: lib/reporters/json-reporter.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ var branchCoverage = function(fileData) {
     for (var j = 0; j < branches.length; j++) {
       if (_.isArray(branches[j])) {
         for (var k = 0; k < branches[j].length; k++) {
-          if (_.isArray(branches[j][k])) {
+          if (_.isObject(branches[j][k])) {
```
Comment (elwayman02) [inline lib/reporters/json-reporter.js]: Why this change? It seems like `branches[j][k]` is being treated as an array on line 39...

File: tests/fixtures/json-output-branches-and-modules.json [hunk 2]
```diff
@@ -1,192 +1,201 @@
...
-      "branchesCovered": 0,
-      "statementsTotal": 4,
-      "statementsCovered": 4,
-      "percentage": 100
-
-    }, {
-      "name": "controllers",
+      "statementsTotal": 113,
+      "statementsCovered": 97,
+      "percentage": 85.84,
```
Comment (elwayman02) [inline tests/fixtures/json-output-branches-and-modules.json]: no files?

File: lib/reporters/json-reporter.js (hunk 1 above)
Comment (yagni) [inline lib/reporters/json-reporter.js]: Good question, and in this reporter you're right--it's being treated as an array. But because blanket stores additional properties (consequent and alternate) on the array it generates at this level, I had to switch this to be an object that looks like an array (its properties are 0, 1, consequent, and alternate) so that those additional properties would be serialized across the wire when posting to the reporter. That way, reporters would have access to them if needed--the HTML reporter in particular uses those additional properties, and others may in the future.

**Comment by yagni** [comment]
Looks like this kind of fell into limbo. Anybody around to review it fully? We'd like to use it for our CI builds.

**Comment by jschilli** [comment]
LGTM

**Comment by clairegrimmett** [comment]
This looks great, we would love to use it too. Any chance of getting this merged?

**Comment by sglanzer** [comment]
Sorry, I've been away from this for a while.  I'm making it a goal to get
some things like this done by the end of the week - stay tuned

**Comment by jschilli** [comment]
I ran this on a few addon projects of ours today.  Merging.
